src/ApiPort.VisualStudio.2017/ApiPort.VisualStudio.2017.csproj

```diff
@@ -29,133 +27,34 @@
-      <HintPath>..\..\packages\System.Collections.Immutable.1.3.1\lib\portable-net45+win8+wp8+wpa81\System.Collections.Immutable.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
-    <Reference Include="System.Composition.AttributedModel, Version=1.0.27.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.Composition.1.0.27\lib\portable-net45+win8+wp8+wpa81\System.Composition.AttributedModel.dll</HintPath>
-    </Reference>
-    <Reference Include="System.Composition.Convention, Version=1.0.27.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.Composition.1.0.27\lib\portable-net45+win8+wp8+wpa81\System.Composition.Convention.dll</HintPath>
-    </Reference>
-    <Reference Include="System.Composition.Hosting, Version=1.0.27.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.Composition.1.0.27\lib\portable-net45+win8+wp8+wpa81\System.Composition.Hosting.dll</HintPath>
-    </Reference>
-    <Reference Include="System.Composition.Runtime, Version=1.0.27.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.Composition.1.0.27\lib\portable-net45+win8+wp8+wpa81\System.Composition.Runtime.dll</HintPath>
-    </Reference>
-    <Reference Include="System.Composition.TypedParts, Version=1.0.27.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.Composition.1.0.27\lib\portable-net45+win8+wp8+wpa81\System.Composition.TypedParts.dll</HintPath>
-    </Reference>
     <Reference Include="System.Core" />
-    <Reference Include="System.Threading.Tasks.Dataflow, Version=4.5.24.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
-      <HintPath>..\..\packages\Microsoft.Tpl.Dataflow.4.5.24\lib\portable-net45+win8+wpa81\System.Threading.Tasks.Dataflow.dll</HintPath>
-      <Private>True</Private>
-    </Reference>
     <Reference Include="System.Xml.Linq" />
     <Reference Include="System.Data.DataSetExtensions" />
     <Reference Include="Microsoft.CSharp" />
     <Reference Include="System.Data" />
-    <Reference Include="System.Net.Http" />
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
     <Folder Include="Properties\" />
   </ItemGroup>
   <ItemGroup>
+    <PackageReference Include="Autofac" Version="4.4.0" />
+    <PackageReference Include="MicroBuild.Core" Version="0.2.0" />
+    <PackageReference Include="Microsoft.Composition" Version="1.0.27" />
+    <PackageReference Include="Microsoft.Tpl.Dataflow" Version="4.5.24" />
+    <PackageReference Include="Microsoft.VisualStudio.ComponentModelHost" Version="14.0.25424" />
+    <PackageReference Include="Microsoft.VisualStudio.Composition" Version="14.2.19-pre" />
```

Comment:
The packages don't match the original ones... so I am getting downgrade errors in `app.config`;
```
<package id="Microsoft.VisualStudio.ComponentModelHost" version="15.0.26201" targetFramework="net46" /> 
<package id="Microsoft.VisualStudio.Composition" version="15.0.70" targetFramework="net46" /> 
```
Same with Microsoft.VisualStudio.Validation.

Reply:
I got that at one point and fixed it. Where are you seeing it?

Reply:
When I'm doing the restore NuGet packages for solution or rebuild in visual studio 2017.

Reply:
```
Restoring NuGet packages...
To prevent NuGet from restoring packages during build, open the Visual Studio Options dialog, click on the Package Manager node and uncheck 'Allow NuGet to download missing packages during build.'
NU1605: Detected package downgrade: Microsoft.VisualStudio.Composition from 15.0.70 to 14.2.19-pre. Reference the package directly from the project to select a different version. 
 ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.ProjectSystem (>= 15.0.751) -> Microsoft.VisualStudio.Composition (>= 15.0.70) 
 ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.Composition (>= 14.2.19-pre)
NU1605: Detected package downgrade: Microsoft.VisualStudio.Validation from 15.0.82 to 14.1.111. Reference the package directly from the project to select a different version. 
 ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.ProjectSystem (>= 15.0.751) -> Microsoft.VisualStudio.Validation (>= 15.0.82) 
 ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.Validation (>= 14.1.111)
NU1605: Detected package downgrade: Microsoft.VisualStudio.Validation from 15.3.15 to 14.1.111. Reference the package directly from the project to select a different version. 
 ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.Threading (>= 15.3.35) -> Microsoft.VisualStudio.Validation (>= 15.3.15) 
 ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.Validation (>= 14.1.111)
NU1605: Detected package downgrade: Microsoft.VisualStudio.ProjectSystem from 15.0.751 to 14.1.170-pre. Reference the package directly from the project to select a different version. 
 ApiPort.Vsix (>= 1.0.0) -> ApiPort.VisualStudio (>= 1.0.0) -> ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.ProjectSystem (>= 15.0.751) 
 ApiPort.Vsix (>= 1.0.0) -> ApiPort.VisualStudio (>= 1.0.0) -> Microsoft.VisualStudio.ProjectSystem (>= 14.1.170-pre)
NU1605: Detected package downgrade: Microsoft.VisualStudio.Threading from 15.3.35 to 14.1.131. Reference the package directly from the project to select a different version. 
 ApiPort.Vsix (>= 1.0.0) -> ApiPort.VisualStudio (>= 1.0.0) -> ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.Threading (>= 15.3.35) 
 ApiPort.Vsix (>= 1.0.0) -> ApiPort.VisualStudio (>= 1.0.0) -> Microsoft.VisualStudio.Threading (>= 14.1.131)
NU1605: Detected package downgrade: Microsoft.VisualStudio.ProjectSystem from 15.0.751 to 14.1.170-pre. Reference the package directly from the project to select a different version. 
 ApiPort.VisualStudio (>= 1.0.0) -> ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.ProjectSystem (>= 15.0.751) 
 ApiPort.VisualStudio (>= 1.0.0) -> Microsoft.VisualStudio.ProjectSystem (>= 14.1.170-pre)
NU1605: Detected package downgrade: Microsoft.VisualStudio.Threading from 15.3.35 to 14.1.131. Reference the package directly from the project to select a different version. 
 ApiPort.VisualStudio (>= 1.0.0) -> ApiPort.VisualStudio.2017 (>= 1.0.0) -> Microsoft.VisualStudio.Threading (>= 15.3.35) 
 ApiPort.VisualStudio (>= 1.0.0) -> Microsoft.VisualStudio.Threading (>= 14.1.131)
1>------ Rebuild All started: Project: Microsoft.Fx.Portability, Configuration: Debug Any CPU ------
1>Microsoft.Fx.Portability -> D:\git\conniey\dotnet-apiport\bin\Debug\Microsoft.Fx.Portability\net46\Microsoft.Fx.Portability.dll
1>Microsoft.Fx.Portability -> D:\git\conniey\dotnet-apiport\bin\Debug\Microsoft.Fx.Portability\netstandard1.3\Microsoft.Fx.Portability.dll
2>------ Rebuild All started: Project: Microsoft.Fx.Portability.Cci, Configuration: Debug Any CPU ------
2>Microsoft.Fx.Portability.Cci -> D:\git\conniey\dotnet-apiport\bin\Debug\Microsoft.Fx.Portability.Cci\netstandard1.3\Microsoft.Fx.Portability.Cci.dll
3>------ Rebuild All started: Project: ApiPort.VisualStudio.Common, Configuration: Debug Any CPU ------
3>  ApiPort.VisualStudio.Common -> D:\git\conniey\dotnet-apiport\bin\Debug\ApiPort.VisualStudio.Common\ApiPort.VisualStudio.Common.dll
4>------ Rebuild All started: Project: ApiPort.VisualStudio.2015, Configuration: Debug Any CPU ------
5>------ Rebuild All started: Project: ApiPort.VisualStudio.2017, Configuration: Debug Any CPU ------
4>  ApiPort.VisualStudio.2015 -> D:\git\conniey\dotnet-apiport\bin\Debug\ApiPort.VisualStudio.2015\ApiPort.VisualStudio.2015.dll
5>C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(1987,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
5>  ApiPort.VisualStudio.2017 -> D:\git\conniey\dotnet-apiport\bin\Debug\ApiPort.VisualStudio.2017\ApiPort.VisualStudio.2017.dll
6>------ Rebuild All started: Project: ApiPort.VisualStudio, Configuration: Debug Any CPU ------
6>C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\MSBuild\15.0\Bin\Microsoft.Common.CurrentVersion.targets(1987,5): warning MSB3277: Found conflicts between different versions of the same dependent assembly that could not be resolved.  These reference conflicts are listed in the build log when log verbosity is set to detailed.
6>CSC : error CS0016: Could not write to output file 'D:\git\conniey\dotnet-apiport\bin\Debug\eoy1ssmx\eoy1ssmx.xml' -- 'Could not find a part of the path 'D:\git\conniey\dotnet-apiport\bin\Debug\eoy1ssmx\eoy1ssmx.xml'.'
7>------ Rebuild All started: Project: ApiPort.Vsix, Configuration: Debug Any CPU ------
7>C:\Users\conniey\.nuget\packages\microsoft.vssdk.buildtools\15.1.192\tools\VSSDK\Microsoft.VsSDK.targets(80,5): warning VSSDK1018: Targets version (1.0.179.29298) and build task version (1.0.249.40350) do not match.  Restarting Visual Studio could help fix this issue.
========== Rebuild All: 5 succeeded, 2 failed, 0 skipped ==========

```

Reply:
Ah - seems that when you have WPF assets, it will create a tmp project for baml generation. Pull latest commit and try building again (do a git clean, though, before)